File: dashlane-project/Dashlane/src/main/java/com/dashlane/login/dagger/LoginCreateAccountSharedModule.java
package com.dashlane.login.dagger;

import com.dashlane.login.TrackingIdProvider;
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactory;
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactoryImpl;
import com.dashlane.login.sso.LoginSsoLoggerModule;

import androidx.fragment.app.FragmentActivity;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.hilt.android.scopes.ActivityScoped;

@Module(includes = LoginSsoLoggerModule.class)
public abstract class LoginCreateAccountSharedModule {
    @Binds
    public abstract ContactSsoAdministratorDialogFactory bindLoginSsoContactAdministratorDialogFactory(
            ContactSsoAdministratorDialogFactoryImpl impl);

    @Provides
    @ActivityScoped
    @TrackingId
    static String getTrackingId(FragmentActivity fragmentActivity) {
        return TrackingIdProvider.INSTANCE.getOrGenerateTrackingId(fragmentActivity);
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/login/dagger/LoginCreateAccountSharedModule.kt
package com.dashlane.login.dagger

import androidx.fragment.app.FragmentActivity
import com.dashlane.login.TrackingIdProvider
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactory
import com.dashlane.login.sso.ContactSsoAdministratorDialogFactoryImpl
import com.dashlane.login.sso.LoginSsoLoggerModule
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@Module(includes = [LoginSsoLoggerModule::class])
@InstallIn(ActivityComponent::class)
abstract class LoginCreateAccountSharedModule {

    @Binds
    abstract fun bindContactSsoAdministratorDialogFactory(factory: ContactSsoAdministratorDialogFactoryImpl): ContactSsoAdministratorDialogFactory

    companion object {
        @Provides
        @TrackingId
        fun getTrackingId(fragmentActivity: FragmentActivity): String = TrackingIdProvider.getOrGenerateTrackingId(fragmentActivity)
    }
}